src/App.tsx

```diff
@@ -15,37 +15,109 @@
   'Garlic',
 ];
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
-
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
-
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
-      </div>
+export enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+interface AppState {
+  isReversed: boolean;
+  sortType: SortType;
+}
+
+export class App extends React.Component<{}, AppState> {
+  state: AppState = {
+    isReversed: false,
+    sortType: SortType.DEFAULT,
+  };
+
+  getSortedGoods = (goods: string[]): string[] => {
+    const { sortType, isReversed } = this.state;
+    const sortedGoods = [...goods];
+
+    if (sortType === SortType.ALPHABET) {
+      sortedGoods.sort();
+    } else if (sortType === SortType.LENGTH) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  sortByAlphabet = (): void => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = (): void => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reverseGoods = (): void => {
+    this.setState(prevState => ({ isReversed: !prevState.isReversed }));
+  };
+
+  resetGoods = (): void => {
+    this.setState({ sortType: SortType.DEFAULT, isReversed: false });
+  };
+
+  render() {
+    const goods = this.getSortedGoods(goodsFromServer);
+    const { sortType, isReversed } = this.state;
+    const isChanged = sortType !== SortType.DEFAULT || isReversed;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info ${sortType === SortType.ALPHABET ? '' : 'is-light'}`}
```

Comment:
Consider using more descriptive class names or comments to clarify the purpose of the conditional class application. This will help maintainability and readability of the code.